src/utils.js

```diff
@@ -236,6 +236,84 @@ function camelcase (dashCaseText) {
     })
 }
 
+// We use this method to detect M1 macs and set appropriate API values to prevent sites from detecting fingerprinting protections
+function isAppleSilicon () {
+    const canvas = document.createElement('canvas')
+    const gl = canvas.getContext('webgl')
+
+    // Best guess if the device is an Apple Silicon
+    // https://stackoverflow.com/a/65412357
+    return gl.getSupportedExtensions().indexOf('WEBGL_compressed_texture_etc') !== -1
+}
+
+/**
+ * Take configSeting which should be an array of possible values.
+ * If a value contains a criteria that is a match for this environment then return that value.
+ * Otherwise return the first value that doesn't have a criteria.
+ *
+ * @param {array} configSetting - Config setting which should contain a list of possible values
+ * @returns - The value from the list that best matches the criteria in the config
+ */
+function processAttrByCriteria (configSetting) {
+    if (!configSetting.some(item => item.criteria !== undefined)) {
```

Comment:
I think this will give out some unexpected behaviour.

You've verified that the argument to this method is an array in the _getFeatureAttr_ callsite, so this code if the items didn't have a _criteria_  property will return _[{...}, {...}]_

I'd delete this statement.